src/QMCDrivers/WFOpt/QMCCostFunctionBase.cpp

```diff
@@ -1058,6 +1063,27 @@ bool QMCCostFunctionBase::isEffectiveWeightValid(EffectiveWeight effective_weigh
   return true;
 }
 
+UniqueOptObjRefs QMCCostFunctionBase::extractOptimizableObjects(TrialWaveFunction& psi) const
+{
+  const auto& names(variational_subset_names);
+  /// survey all the optimizable objects
+  const auto opt_obj_refs = psi.extractOptimizableObjectRefs();
+  for (OptimizableObject& obj : opt_obj_refs)
+    obj.setOptimization(names.empty() || std::find_if(names.begin(), names.end(), [&](const std::string& name) {
```

Comment:
This took me a while to parse out, I think it would be easier to read (and cleaner) without the  `&` capture parameter.  Just use `[&obj]`